### PR TITLE
Add First Draft of Lint Listing Page

### PR DESCRIPTION
**Pull request by killercup**
cf. #1088

- export.py is 80% copy-pasta from [update_wiki.py](https://github.com/Manishearth/rust-clippy/blob/b14114f2537b06731b3be4ecbeea1e4c47d3eb12/util/update_wiki.py)
- index.html is 70% copy-pasta from [bitrust](https://github.com/killercup/bitrust/blob/208374ab8fb8d104e3934fbdfebc21df802bc7c9/index.html)

Test:

clone this branch and run

```sh
$ python util/export.py
$ cd util/gh-pages
$ python -m SimpleHTTPServer 8080
```

### CODE REVIEW

**Comment by killercup** [comment]
My lunch break is over now, so I haven't

- made any custom design adjustments
- set this up to automatically deploy on travis (maybe copy [this](https://github.com/rust-lang-nursery/regex/blob/d95d021a7f150a5b513af166f044ba0c51aa31fb/.travis.yml#L41-L44))

File: .gitignore [hunk 1]
```diff
@@ -16,3 +16,6 @@ Cargo.lock
 
 # Generated by dogfood
 /target_recur/
+
+# hg pages docs
```
Comment (llogiq) [inline .gitignore]: Didn't you mean gh pages?

Comment (killercup) [inline .gitignore]: haha, yes

File: util/export.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                if "}" in l:
+                    warn("Warning: Missing Lint-Name in", f)
+                    comment = True
+
+def main():
+    (lints, config) = parse_path()
+    info("got %s lints" % len(lints))
+    with open("util/gh-pages/lints.json", "w") as file:
+        json.dump(lints, file, indent=2)
+        info("wrote JSON for greate justice")
```
Comment (llogiq) [inline util/export.py]: great! But not greate. :smile:

Comment (killercup) [inline util/export.py]: Yes, I just fixed that useless debug message :D

**Comment by killercup** [comment]
Anything else (more typos?) you want me to change here?

I can't test the travis integration, so someone with the ability to push to this repo needs to create an API key and adjust the travis config. Then we can get this on the road :)

**Comment by mcarton** [comment]
Some comments:
* You should `chmod +x util/export.py`.
* All lints `<article>` should have an html `id` so that we can point to it as we currently do with the wiki.
* The file has some space-only lines.
* The level selector should have `Warn + Deny` option since that's what user will have by default.
* The html file has a “Error loading commits!” message (bad copy-pasta :smile:).
* Some lints have redundant sections:

    ### Example

    Example:

    ```rust
    let mut a = 5;
    ...
    a = a + b;
    ```
* How hard would it be to have syntax highlighting (is the markdown rendered client side? :confused:) ?
* I'd like an “open/close all” option (bonus).
* :+1:

**Comment by mcarton** [comment]
* The table in the `ineffective_bit_mask` is not rendered.

**Comment by mcarton** [comment]
* The font size of citations is much bigger than regular text (`linkedlist` looks really weird).

**Comment by killercup** [comment]
I think I fixed everything you mentioned, @mcarton, except for the open all/close all feature (it's currently really lazy, and thus harder to change all values at once).

I have to say, I'm both surprised and shocked by how efficient implementing something like this in angular.js is. The code is anything but nice/performant/zen, but it works 😄

**Comment by mcarton** [comment]
There is still the problem with `unnecessary_mut_passed`, `used_underscore_binding` and `wrong_self_convention`.

Also why is the inline code red?
screen shot 2016-07-14 at 21 35 33

Otherwise, LGTM.

**Comment by mcarton** [comment]
> this might also be a problem when generating the wiki page!

[It looks fine](https://github.com/Manishearth/rust-clippy/wiki#unnecessary_mut_passed)

> That's Bootstrap's default style. Do you want me to make it grayscale instead?

I though it was weird, but if it's Bootstrap leave it.

**Comment by mcarton** [comment]
@killercup another thing: configuration variables do not appear in your doc (they do in the wiki, see eg. [`blacklisted_name`](https://github.com/Manishearth/rust-clippy/wiki#blacklisted_name)).

**Comment by killercup** [comment]
> There is still the problem with unnecessary_mut_passed, used_underscore_binding and wrong_self_convention.

My regex was too good. These lint docs say "Example", not "Example:" (with the colon at the end) 😄 

> I though it was weird, but if it's Bootstrap leave it.

Haha, I did not realize saying "Bootstrap" works as an argument from authority!

> configuration variables do not appear in your doc

Oh, they were in an entirely different dict that I did not merge with the lint data _at all_. Fixed that. Also learned some python tricks along the way!

**Comment by killercup** [comment]
Anything else I need to fix? Someone still needs to add and test the travis config.

**Comment by mcarton** [comment]
I'm fine with this.
I was going to deploy it just to realize I don't have access to <https://github.com/Manishearth/rust-clippy/settings/keys>. @Manishearth: it looks like you'll have to had a deploy key yourself for Travis.

**Comment by Manishearth** [comment]
Huh, there's no way to give you access either. hold on.

**Comment by Manishearth** [comment]
Emailed a deploy key. I assume you're able to run travis encrypt locally?

**Comment by mcarton** [comment]
@killercup just realized when you have a filter and click a link to a lint that's filtered out, nothing happens. Is there some JS voodoo magic you could had here? :smile:

Eg.:
* Go on https://manishearth.github.io/rust-clippy/current/
* Set filter “break”
* Open “wrong_pub_self_convention”
* Click on the first link: “This is the same as [`wrong_self_convention`](https://manishearth.github.io/rust-clippy/current/#wrong_self_convention), but for public items.”
⇒ Nothing happens.

**Comment by killercup** [comment]
Oh, very interesting, @mcarton! I hadn't tested the inter-lint-links at all (didn't even know they existed tbh). Can you make that into a new issue and assign it to me, so I don't forget to fix this? :)

**Comment by mcarton** [comment]
That one exists since https://github.com/Manishearth/rust-clippy/pull/1146, there might have been some other but that's a good idea IMO.

**Pull request merged**